pkg/queue/concurrency_state.go

```diff
@@ -126,9 +124,19 @@ func NewConcurrencyEndpoint(e, m string) ConcurrencyEndpoint {
 	return c
 }
 
-func (c ConcurrencyEndpoint) Pause() error { return c.Request("pause") }
+func (c ConcurrencyEndpoint) Pause(logger *zap.SugaredLogger) error {
+	if err := c.Request("pause"); err != nil {
+		handleStateRequestError(logger, c.Pause)
+	}
+	return nil
```

Comment:
The pattern here is a bit "weird" to me, as we're essentially trying a first call and then handle the retries as "special" calls. We're also calling into the retry logic recursively by recalling `c.Pause` here.

Instead, what I'd expect is a call like `retryingRequest(logger, c.Request("pause")`. The implementation can be almost identical with `handleStateRequestError`, but I'd call `wait.PollImmediate` so that the first call is made immediately, without a wait.

That'd normalize the flow to be **always** via the same path and avoid the recursive call.